### PR TITLE
Enhance SAC with loss function and subsampling logic

### PR DESCRIPTION
**Pull request by Ke-Wang1017**
## What this does
I modified especially the loss computation part because the existing one has some mismatches with the network definition. More specifically:

- Added `num_subsample_critics`, `critic_target_update_weight`, and `utd_ratio` to SACConfig.
- Implemented target entropy calculation in SACPolicy if not provided.
- Introduced subsampling of critics to prevent overfitting during updates.
- Updated temperature loss calculation to use the new target entropy.
- Added comments for future UTD update implementation.

### CODE REVIEW

**Comment by Ke-Wang1017** [comment]
Can you please have a look @michel-aractingi

File: lerobot/scripts/control_robot.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def record(
     else:
         raise NotImplementedError("Only single-task recording is supported for now")
 
-    # Load pretrained policy
+        # Load pretrained policy
```
Comment (michel-aractingi) [inline lerobot/scripts/control_robot.py:227]:
```suggestion
    # Load pretrained policy
```
wrong indentation?

**Comment by michel-aractingi** [comment]
Closing as the commits were added manually to avoid conflicts in commit history issue.
